pkg/ebpf/telemetry/modifier.go

```diff
@@ -22,16 +38,105 @@ func (t *ErrorsTelemetryModifier) String() string {
+	if !slices.ContainsFunc(m.Maps, func(x *manager.Map) bool { return x.Name == mapErrTelemetryMapName }) {
+		m.Maps = append(m.Maps, &manager.Map{Name: mapErrTelemetryMapName})
+	}
+	if !slices.ContainsFunc(m.Maps, func(x *manager.Map) bool { return x.Name == helperErrTelemetryMapName }) {
+		m.Maps = append(m.Maps, &manager.Map{Name: helperErrTelemetryMapName})
+	}
+
+	// set a small max entries value if telemetry is not supported. We have to load the maps because the eBPF code
+	// references them even when we cannot track the telemetry.
+	opts.MapSpecEditors[mapErrTelemetryMapName] = manager.MapSpecEditor{
+		MaxEntries: uint32(1),
+		EditorFlag: manager.EditMaxEntries,
+	}
+	opts.MapSpecEditors[helperErrTelemetryMapName] = manager.MapSpecEditor{
+		MaxEntries: uint32(1),
+		EditorFlag: manager.EditMaxEntries,
+	}
+
+	if activateBPFTelemetry {
+		collectionSpec, err := ebpf.LoadCollectionSpecFromReader(bytecode)
+		if err != nil {
+			return fmt.Errorf("failed to load collection spec for module %s: %w", module.String(), err)
+		}
+
+		opts.MapSpecEditors[mapErrTelemetryMapName] = manager.MapSpecEditor{
+			MaxEntries: uint32(len(collectionSpec.Maps)),
+			EditorFlag: manager.EditMaxEntries,
+		}
+		log.Tracef("module %s maps %d", module.String(), opts.MapSpecEditors[mapErrTelemetryMapName].MaxEntries)
+
+		opts.MapSpecEditors[helperErrTelemetryMapName] = manager.MapSpecEditor{
+			MaxEntries: uint32(len(collectionSpec.Programs)),
+			EditorFlag: manager.EditMaxEntries,
+		}
+		log.Tracef("module %s probes %d", module.String(), opts.MapSpecEditors[helperErrTelemetryMapName].MaxEntries)
+
+		h := keyHash()
+		for _, ebpfMap := range m.Maps {
+			opts.ConstantEditors = append(opts.ConstantEditors, manager.ConstantEditor{
+				Name:  ebpfMap.Name + "_telemetry_key",
+				Value: eBPFMapErrorKey(h, mapTelemetryKey(names.NewMapNameFromManagerMap(ebpfMap), module)),
+			})
+		}
+
+	}
+
+	m.InstructionPatchers = append(m.InstructionPatchers, func(m *manager.Manager) error {
+		specs, err := m.GetProgramSpecs()
+		if err != nil {
 			return err
 		}
+		return patchEBPFTelemetry(specs, activateBPFTelemetry, module, errorsTelemetry)
+	})
+
+	return nil
+}
+
+// AfterInit pre-populates the telemetry maps with entries corresponding to the ebpf program of the manager.
+func (t *ErrorsTelemetryModifier) AfterInit(m *manager.Manager, module names.ModuleName, _ *manager.Options) error {
+	if errorsTelemetry == nil {
+		return nil
+	}
+
+	var mapNames []names.MapName
+	for _, mp := range m.Maps {
+		mapNames = append(mapNames, names.NewMapNameFromManagerMap(mp))
+	}
+
+	mapErrMap, _, err := m.GetMap(mapErrTelemetryMapName)
+	if err != nil {
+		return fmt.Errorf("failed to find map %s: %w", mapErrTelemetryMapName, err)
+	}
+	genericMapErrMap, err := maps.Map[uint64, mapErrTelemetry](mapErrMap)
+	if err != nil {
+		return fmt.Errorf("failed to create generic map from map spec for %s: %w", mapErrTelemetryMapName, err)
+	}
```

Comment:
There's a shorthand in the `maps` package for this:

```suggestion
	genericMapErrMap, err := maps.GetMap[uint64, mapErrTelemetry](m, mapErrTelemetryName)
	if err != nil {
		return fmt.Errorf("failed to get map %s: %w", mapErrTelemetryMapName, err)
	}
```